### DISCUSSION TITLE
Setting Up LocoNodes for Larger Area Coverage (Like Skybrush's Big Loco Demo)

### DISCUSSION THREAD

**Discussion by amitx64** [bitcraze/discussions#1635] [Q&A]
Hi,

I’m trying to set up `LocoNodes` for a larger area, aiming to create a setup similar to the `Big Loco Demo` done in collaboration with `Skybrush` and `Bitcraze`. 

I asked on Discord for some guidance, and I was advised to reach out here since some of your team members who might know the details aren’t on Discord. Could you provide insights on the specific parameters or configurations required to extend the area effectively?

**Comment by Bananazooka** [comment]
Hi, you can find details on how we did it [here](https://www.bitcraze.io/2022/12/the-big-loco-show/).

**Reply by amitx64** [reply]
@Bananazooka But there is `no information` regarding the `power of the Loco node` in a `larger area` in the `Big Loco demo test`.

**Comment by Bananazooka** [comment]
Hi, what do you mean with the power of the Loco nodes?

**Reply by amitx64** [reply]
@Bananazooka, I noticed that `Bitcraze` placed the anchors over a wide area, with approximately 12 meters of spacing between consecutive Loco nodes in `Big Loco Test`. During that setup, did `Bitcraze increase the power of the Loco nodes`, or were they `operating at the default setting of 3.5 dB`?

**Reply by amitx64** [reply]
@Bananazooka @ataffanel @tobbeanton, to `mitigate z-variation`, if I mount the `zRanger2` on the Crazyflie along with the `UWB deck`, will it work together with the UWB? Also, if the `zRanger2` does work, what will happen when the drone flies above 4 meters? As you know, the `zRanger2` only works up to `4 meters`.

**Reply by amitx64** [reply]
@Bananazooka @ataffanel @tobbeanton, Does the Crazyflie handle `dynamic switching` between the `zRanger2` and the `Loco Positioning deck` for z-position measurement when transitioning above or below 4 meters, or do I need to implement that myself?

**Reply by amitx64** [reply]
@Bananazooka @ataffanel @tobbeanton, If I want to implement this `dynamic switching` between the `zRanger2` and the `Loco Positioning deck` for `z-position measurement` when `transitioning above or below 4 meters`, then where I need to change?

**Reply by knmcguire** [reply]
@amitx64 Please do not attach all of these github handles in multiple messages. This is quite disruptive in our notifications.


For your questions these are very specific and will require quite some development on your part. We can't help you every step of the way as of these weeks we will be running with a skeleton crew at the office but I'll refer you to the right documentation:

* To increase the TX power (which is what is referred to), you can find the instructions to do that in the LPS-node-firmware documentation: https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/
* To check if the z-ranger is compatible with the loco deck you can chec the expansion deck documentation: https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/
* No there is no dynamic switching. You'll need to probably implement some kind of filter in the z-ranger driver in the[ Crazyflie firmware](https://github.com/bitcraze/crazyflie-firmware). It should be in src/deck/drivers.

**Reply by amitx64** [reply]
I'm sorry about that. That wasn't my intention.

**Reply by amitx64** [reply]
And thank you for the information. I will be going to tell my test result.

**Reply by amitx64** [reply]
When I tested the `zranger2` deck together with the `Loco Positioning System (LPS)` deck, I observed that the Ultra-Wideband (UWB) position signal becomes unreliable or is lost entirely once the drone moves approximately 1.1 meters away. It’s also possible that the continuous movement of the drone might be contributing to this loss of position data. Additionally, I noticed that the altitude seems to decrease unexpectedly after this point. I am not entirely sure about the cause of these issues, but I suspect it may be related to the frequency or timing of the `UWB task`.

Would adjusting the `UWB task` frequency be helpful in improving positioning stability? Maybe increasing the update rate could help maintain more consistent position data during drone movement. What are your thoughts on adjusting the UWB task frequency or possibly synchronizing the updates with the `zranger2` altitude readings?

**Reply by knmcguire** [reply]
Before going into the timing, I would advise you to also look at the raw measurements of the height and ranges as well. The kalman filter will diverge in case there is 1. no measurement or 2. very faulty measurement. 

The [SDcard ](https://www.bitcraze.io/products/micro-sd-card-deck/)deck will allow you to log the raw measurement data at a very fast rate.

**Reply by amitx64** [reply]
Ok got it
